ci/benchmark/build.sh

```diff
@@ -0,0 +1,191 @@
+# conda install "your-pkg=1.0.0"
+
+# Install the master version of dask, distributed, and streamz
+logger "pip install git+https://github.com/dask/distributed.git --upgrade --no-deps"
+pip install "git+https://github.com/dask/distributed.git" --upgrade --no-deps
+logger "pip install git+https://github.com/dask/dask.git --upgrade --no-deps"
+pip install "git+https://github.com/dask/dask.git" --upgrade --no-deps
+logger "pip install git+https://github.com/python-streamz/streamz.git --upgrade --no-deps"
+pip install "git+https://github.com/python-streamz/streamz.git" --upgrade --no-deps
+
+logger "Check versions..."
+python --version
+$CC --version
+$CXX --version
+conda list
+
+################################################################################
+# BUILD - Build libcudf, cuDF and dask_cudf from source
+################################################################################
+
+logger "Build libcudf..."
+if [[ ${BUILD_MODE} == "pull-request" ]]; then
+    $WORKSPACE/build.sh clean libcudf cudf dask_cudf benchmarks tests --ptds
+else
+    $WORKSPACE/build.sh clean libcudf cudf dask_cudf benchmarks tests -l --ptds
+fi
+
+################################################################################
+# BENCHMARK - Run and parse libcudf and cuDF benchmarks
+################################################################################
+
+logger "Running benchmarks..."
+
+#Download GBench results Parser
+curl -L https://raw.githubusercontent.com/rapidsai/benchmark/main/parser/GBenchToASV.py --output GBenchToASV.py
+
+###
+# Generate Metadata for dependencies
+###
+
+# Concatenate dependency arrays, convert to JSON array,
+# and remove duplicates.
+X=("${CUDF_DEPS[@]}" "${LIBCUDF_DEPS[@]}")
+DEPS=$(printf '%s\n' "${X[@]}" | jq -R . | jq -s 'unique')
+
+# Build object with k/v pairs of "dependency:version"
+DEP_VER_DICT=$(jq -n '{}')
+for DEP in $(echo "${DEPS}" | jq -r '.[]'); do
+  VER=$(conda list | grep "^${DEP}" | awk '{print $2"-"$3}')
+  DEP_VER_DICT=$(echo "${DEP_VER_DICT}" | jq -c --arg DEP "${DEP}" --arg VER "${VER}" '. + { ($DEP): $VER }')
+done
+
+# Pass in an array of dependencies to get a dict of "dependency:version"
+function getReqs() {
+  local DEPS_ARR=("$@")
+  local REQS="{}"
+  for DEP in "${DEPS_ARR[@]}"; do
+    VER=$(echo "${DEP_VER_DICT}" | jq -r --arg DEP "${DEP}" '.[$DEP]')
+    REQS=$(echo "${REQS}" | jq -c --arg DEP "${DEP}" --arg VER "${VER}" '. + { ($DEP): $VER }')
+  done
+
+  echo "${REQS}"
+}
+
+###
+# Run LIBCUDF Benchmarks
+###
+
+REQS=$(getReqs "${LIBCUDF_DEPS[@]}")
+
+mkdir -p ${WORKSPACE}/tmp/benchmark
+touch ${WORKSPACE}/tmp/benchmark/benchmarks.txt
+ls ${GBENCH_BENCHMARKS_DIR} > ${WORKSPACE}/tmp/benchmark/benchmarks.txt
+
+#Disable error aborting while tests run, failed tests will not generate data
+logger "Running libcudf GBenchmarks..."
+cd ${GBENCH_BENCHMARKS_DIR}
+set +e
+while read BENCH;
+do
+    nvidia-smi
+    ./${BENCH} --benchmark_out=${BENCH}.json --benchmark_out_format=json
```

Comment:
If it's part of CI, adding `--benchmark_min_time` option is recommended. See https://github.com/rapidsai/cudf/pull/5744#discussion_r459853030